pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -265,7 +276,8 @@ def _do_check_save(self, filepath, current, epoch):
             self.kth_value = self.best_k_models[self.kth_best_model]
 
         _op = min if self.mode == 'min' else max
-        self.best = _op(self.best_k_models.values())
+        self.best_model = _op(self.best_k_models, key=self.best_k_models.get)
+        self.best = self.best_k_models[self.best_model]
```

Comment:
here the best means `best_score`? otherwise, it is a bit confusing to have `best` and `best_model`
pls add types to the init :rabbit:

Reply:
Yes, `self.best` is the best "monitored quantity". But that's existing naming. Changing it will break backwards compatibility. If that's OK, I can surely rename it. Otherwise, `best_model` could be `best_model_path`. I used `best_model` to keep consistency with the existing `kth_best_model` attribute.

As for type hints, there's no changes to the arguments. Where specifically would I add them?

Reply:
@PyTorchLightning/core-contributors are we fine to rename `best` >> `best_score`?

Reply:
i would prefer it (`best_score`), but if we do that we need to provide our users with a utility to upgrade their checkpoints (basically switch `best` to `best_score`) and temporarily have a warning which catches old checkpoints at load time and alerts users of the utility to convert. it's a very simple utility but we should provide it for our users if we move forward with this.

Reply:
@jeremyjordan you mean to keep compatible with past saved checkpoint which contains `best`? That shall be simple, in loading, we will map `best` >> `best_score`... and yes we shall wrap the deprecated `best` with some warning :]

Reply:
OK, I renamed `best` to `best_model_score` and `best_model` to `best_model_path`, since having `best_score` and `best_model` would still be a bit confusing (IMHO). To keep consistency, I also renamed `kth_best_model` to `kth_best_model_path`.

I added properties for `best` and `kth_best_model` that log a deprecation warning and return the correct value.

When loading a checkpoint, if it's in an old format, the value for `best` is simply assigned to `best_model_score`. In my opinion, adding a warning in this part will not really help the user, as there's not much they can do.

Let me know of any further changes.